### PR TITLE
Guard bg[i] from being null in render BG graph

### PR DESCRIPTION
**Pull request by jaylagorio**
I was getting this error during looping and it would stop the loop:

```
/root/src/openaps-menu/scripts/status.js:103
    var x = 2 + zero_x + Math.round(((((bg[i].date - zerotime)/1000)/60)/5));
                                              ^

TypeError: Cannot read property 'date' of undefined
    at Object.<anonymous> (/root/src/openaps-menu/scripts/status.js:103:47)
    at Module._compile (module.js:652:30)
    at Object.Module._extensions..js (module.js:663:10)
    at Module.load (module.js:565:32)
    at tryModuleLoad (module.js:505:12)
    at Function.Module._load (module.js:497:3)
    at Function.Module.runMain (module.js:693:10)
    at startup (bootstrap_node.js:188:16)
    at bootstrap_node.js:609:3
```

This led me to believe that one of the items in the array is null, so this skips that loop iteration. After doing this the loop was running properly again.

### CODE REVIEW

File: scripts/status.js [hunk 1]
```diff
@@ -100,18 +100,20 @@ var date = new Date();
 var zerotime = date.getTime() - ((numBGs * 5) * 600);
 var zero_x = numBGs + 5;
 for (var i = 0; i <= numBGs; i++) {
```
Comment (scottleibrand) [inline scripts/status.js]: The underlying bug is this `<=` should be a `<`.  Want to fix that as well (and leave your null check too)?

**Comment by jaylagorio** [comment]
Nice catch! Change made.

**Comment by scottleibrand** [comment]
There was a big long discussion in gitter about it, but apparently no one actually PR'd the fix, so thanks for doing that.  :-)

**Comment by jaylagorio** [comment]
I've been so absent from gitter I missed it entirely. You're welcome, thanks for the merge!

**Comment by cluckj** [comment]
:-P I fixed it in https://github.com/openaps/openaps-menu/pull/13

**Comment by scottleibrand** [comment]
Ah, sorry. Is this change compatible with that one, or should I revert?

**Comment by cluckj** [comment]
It's incompatible, but it has the same end result so :man_shrugging: I'll change #13 to fix the display offset if it still needs it, or close it if it doesn't.

**Pull request merged**